onnxruntime/core/providers/rocm/rocm_provider_factory.cc

```diff
@@ -180,21 +194,32 @@ struct ROCM_Provider : Provider {
     return std::make_shared<ROCMProviderFactory>(info);
   }
 
+  /**
+   * This function will be called by the C API UpdateROCMProviderOptions().
+   *
+   * What this function does is equivalent to resetting the OrtROCMProviderOptions instance with
+   * default ROCMExecutionProviderInf instance first and then set up the provided provider options.
+   * See ROCMExecutionProviderInfo::FromProviderOptions() for more details.
+   */
   void UpdateProviderOptions(void* provider_options, const ProviderOptions& options) override {
-    auto info = onnxruntime::ROCMExecutionProviderInfo::FromProviderOptions(options);
+    auto internal_options = onnxruntime::ROCMExecutionProviderInfo::FromProviderOptions(options);
     auto& rocm_options = *reinterpret_cast<OrtROCMProviderOptions*>(provider_options);
 
-    rocm_options.device_id = info.device_id;
-    rocm_options.gpu_mem_limit = info.gpu_mem_limit;
-    rocm_options.arena_extend_strategy = static_cast<int>(info.arena_extend_strategy);
-    rocm_options.miopen_conv_exhaustive_search = info.miopen_conv_exhaustive_search;
-    rocm_options.do_copy_in_default_stream = info.do_copy_in_default_stream;
-    rocm_options.has_user_compute_stream = info.has_user_compute_stream;
-    rocm_options.user_compute_stream = info.user_compute_stream;
-    rocm_options.default_memory_arena_cfg = info.default_memory_arena_cfg;
-    rocm_options.tunable_op_enable = info.tunable_op.enable;
-    rocm_options.tunable_op_tuning_enable = info.tunable_op.tuning_enable;
-    rocm_options.tunable_op_max_tuning_duration_ms = info.tunable_op.max_tuning_duration_ms;
+    rocm_options.device_id = internal_options.device_id;
+    rocm_options.gpu_mem_limit = internal_options.gpu_mem_limit;
+    rocm_options.arena_extend_strategy = static_cast<int>(internal_options.arena_extend_strategy);
+    rocm_options.miopen_conv_exhaustive_search = internal_options.miopen_conv_exhaustive_search;
+    rocm_options.do_copy_in_default_stream = internal_options.do_copy_in_default_stream;
+    rocm_options.has_user_compute_stream = internal_options.has_user_compute_stream;
+    // The 'has_user_compute_stream' of the OrtROCMProviderOptions instance can be set byC API UpdateROCMProviderOptionsWithValue() as well.
```

Comment:
```suggestion
    // The 'has_user_compute_stream' of the OrtROCMProviderOptions instance can be set by C API UpdateROCMProviderOptionsWithValue() as well.
```
maybe sync backward?

Reply:
fixed by 4d93107b6c4c705723ef1b7a4060bc5cf7487d78.